crates/turborepo-lib/src/daemon/server.rs

```diff
@@ -255,6 +258,9 @@ impl<T: Watcher + Send + 'static> proto::turbod_server::Turbod for DaemonServer<
     ) -> Result<tonic::Response<proto::NotifyOutputsWrittenResponse>, tonic::Status> {
         let inner = request.into_inner();
 
+        self.times_saved
+            .insert(inner.hash.clone(), inner.time_saved);
+
```

Comment:
Now the we have this behind a mutex we need to lock it before we can access the data. We introduce a block so that `times_saved` drops as soon as possible. `times_saved` isn't actually the `HashMap` type, but a `MutexGuard` which will release the lock as soon as it gets dropped freeing up the resource for others.
One question I have about this code is what happens if we already have a time saved for an output hash? Do we always want to overwrite the old one?
```suggestion
        {
            let mut times_saved = self.times_saved.lock().expect("times saved lock poisoned");
            times_saved.insert(inner.hash.clone(), inner.time_saved);
        }
```

Reply:
yes we'd always want to overwrite since every time the cache is written and `notifyOutputsWritten` is called the duration is saved in the cache metadata also https://github.com/vercel/turbo/blob/d51ad5a9c775d3d0721e3a560b81f3829e447be1/cli/internal/runcache/runcache.go#L279-L282